google_guest_agent/osinfo/osinfo_unix.go

```diff
@@ -12,42 +12,22 @@
 //  See the License for the specific language governing permissions and
 //  limitations under the License.
 
-package main
+//go:build unix
+
+package osinfo
 
 import (
-	"fmt"
+	"bytes"
 	"io/ioutil"
-	"runtime"
 	"strconv"
 	"strings"
-)
-
-type ver struct {
-	major, minor, patch, length int
-}
-
-// release holds Linux distribution release information.
-type release struct {
-	os      string
-	version ver
-}
 
-func (v ver) String() string {
-	if v.major == 0 {
-		return ""
-	}
-	ret := fmt.Sprintf("%d", v.major)
-	if v.length > 1 {
-		ret = fmt.Sprintf("%s.%d", ret, v.minor)
-	}
-	if v.length > 2 {
-		ret = fmt.Sprintf("%s.%d", ret, v.patch)
-	}
-	return ret
-}
+	"github.com/GoogleCloudPlatform/guest-logging-go/logger"
+	"golang.org/x/sys/unix"
+)
 
-func parseOSRelease(osRelease string) (release, error) {
-	var ret release
+func parseOSRelease(osRelease string) OSInfo {
```

Comment:
These functions here where you are handling errors internally (meaning not returning error anymore), I'd like to have them reporting the error and allowing the caller decide what to do with it, even if it's just logging and not aborting, as a "specialized code" it shouldn't know about the "app"/guest-agent flow.

Reply:
meaning, propagating the error up to the "core" guest-agent code.

Reply:
These are private now, very different from before so I can't pass the errors up to core code. I did move the error handling to the Get function though so that we can test the error cases here.

I could make Get a bit more complicated and return concatenated errors (Get won't return immediately on error, but instead get adding to the potential errors) if you would prefer the caller of Get to handle the logic